### PR TITLE
implement 3 types of funbutton (primary, secondary, tertiary) + Rrequired amplitude tracking

### PR DESCRIPTION
**Pull request by MaikelStuivenberg**
## Summary by CodeRabbit

- **New Features**
	- Enhanced analytics tracking across various user interactions, including login, signup, and button clicks.
	- Introduced new events for improved tracking of user actions within the application.
	- Integrated analytics functionality into multiple components, allowing for better insights into user behavior.

- **Bug Fixes**
	- Corrected event names for consistency in analytics tracking.

- **Refactor**
	- Replaced direct event logging with structured `AnalyticsEvent` objects for better maintainability and clarity.
	- Unified button implementations to improve consistency across the application.

- **Style**
	- Adjusted button styling and component usage for improved visual consistency.

### CODE REVIEW

File: lib/features/children/generosity_challenge/assignments/day4_timer/pages/day4_timer_screen.dart [hunk 1]
```diff
@@ -180,15 +178,18 @@ class _Day4TimerScreenState extends State<Day4TimerScreen> {
...
-                            if (_appConfig.isTestApp)
-                              FunButton(
-                                onTap: () {
-                                  setState(() {
-                                    _remainingSeconds = 20;
-                                  });
-                                },
-                                text: 'Debug: set to 20 seconds remaining',
+                            FunButton(
+                              isDebugOnly: true,
```
Comment (Daniela510) [inline lib/features/children/generosity_challenge/assignments/day4_timer/pages/day4_timer_screen.dart:182]: nice addition 'isDebugOnly'! Its from previous prs but i just noticed, nice!

File: lib/features/family/features/profiles/screens/profile_selection_screen.dart [hunk 2]
```diff
@@ -165,17 +166,17 @@ class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
                             const SizedBox(height: 8),
                             if (state.profiles.length >= 3) ...[
                               FunButton(
-                                isTertiary: true,
                                 onTap: () => context.goNamed(
```
Comment (Daniela510) [inline lib/features/family/features/profiles/screens/profile_selection_screen.dart:169]: why no? 😢

Comment (MaikelStuivenberg) [reply lib/features/family/features/profiles/screens/profile_selection_screen.dart:169]: It's now FunButton.tertiary() and FunButton.secondary() :)

File: lib/features/children/overview/widgets/no_children_page.dart [hunk 3]
```diff
@@ -27,6 +29,9 @@ class NoChildrenPage extends StatelessWidget {
           FunButton(
             onTap: onAddNewChildPressed,
             text: context.l10n.plusAddMembers,
+            analyticsEvent: AnalyticsEvent(
+              AmplitudeEvents.addMemerClicked,
+            ),
```
[the remark this answers is not in the document]
Comment (MaikelStuivenberg) [reply lib/features/children/overview/widgets/no_children_page.dart]: @coderabbitai solved in the latest commit! :)

**Review by Daniela510 — APPROVED** [review]

**Pull request merged**